clickhouse/base/wire_format.cpp

```diff
@@ -0,0 +1,105 @@
+#include "wire_format.h"
+
+#include "input.h"
+#include "output.h"
+
+#include <stdexcept>
+
+namespace {
+static const int MAX_VARINT_BYTES = 10;
+}
+
+namespace clickhouse {
+
+bool WireFormat::ReadAll(InputStream * input, void* buf, size_t len) {
+    uint8_t* p = static_cast<uint8_t*>(buf);
+
+    size_t read_previously = 1; // 1 to execute loop at least once
+    while (len > 0 && read_previously) {
+        read_previously = input->Read(p, len);
+
+        p += read_previously;
+        len -= read_previously;
+    }
+
+    // true if all was read successfully
+    return !len;
+//    if (len) {
+//        throw std::runtime_error("Failed to read " + std::to_string(original_len)
+//                + " bytes, only read " + std::to_string(original_len - len));
+//    }
+}
+
+void WireFormat::WriteAll(OutputStream* output, const void* buf, size_t len) {
+    const size_t original_len = len;
+    const uint8_t* p = static_cast<const uint8_t*>(buf);
+
+    size_t written_previously = 1; // 1 to execute loop at least once
+    while (len > 0 && written_previously) {
+        written_previously = output->Write(p, len);
+
+        p += written_previously;
+        len -= written_previously;
+    }
+
+    if (len) {
+        throw std::runtime_error("Failed to write " + std::to_string(original_len)
+                + " bytes, only written " + std::to_string(original_len - len));
+    }
+}
+
+bool WireFormat::ReadVarint64(InputStream* input, uint64_t* value) {
+    *value = 0;
+
+    for (size_t i = 0; i < MAX_VARINT_BYTES; ++i) {
+        uint8_t byte;
```

Comment:
Uninitialized.

Reply:
yep, it is overwritten right afterward and there is no reason to initialize it.